src/loading.jl

```diff
@@ -29,13 +36,17 @@ function parse_pkg_files(id::PkgId)
     end
     modsym = Symbol(id.name)
     if use_compiled_modules()
-        cachefile, includes, reqs = pkg_fileinfo(id)
+        cachefile, includes, reqs, buildid = pkg_fileinfo(id)
         if cachefile !== nothing
             @assert includes !== nothing
             @assert reqs !== nothing
             pkgdata.requirements = reqs
             for chi in includes
-                mod = Base.root_module(id)
+                if VERSION >= v"1.12.0-DEV.764" && haskey(Base.loaded_precompiles, id => buildid)
```

Comment:
@vtjnash is this the expected fix for this?

Reply:
That probably depends upon what the code is attempting to do. It is supposed to be buildid specific or not?

Reply:
This code is adding the files that a .ji file depends on to Revise's watch list, so I believe it is buildid specific.

Reply:
Ensuring that it uses the same .ji file is probably correct then. We might need to examine whether the package hooks should provide that additional information also, to ensure Revise is accurate about it though?

Reply:
Alright, good enough for now then.